plugin/trino-hive-hadoop2/src/test/java/io/trino/plugin/hive/s3select/TestHiveFileSystemS3SelectPushdownWithSplits.java

```diff
@@ -0,0 +1,110 @@
+import io.trino.plugin.hive.HiveConfig;
+import io.trino.spi.connector.ColumnHandle;
+import io.trino.spi.connector.ConnectorSession;
+import io.trino.spi.connector.SchemaTableName;
+import io.trino.testing.MaterializedResult;
+import org.apache.hadoop.fs.Path;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Parameters;
+import org.testng.annotations.Test;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.LongStream;
+
+import static io.airlift.units.DataSize.Unit.MEGABYTE;
+import static io.trino.plugin.hive.HiveColumnHandle.ColumnType.REGULAR;
+import static io.trino.plugin.hive.HiveColumnHandle.createBaseColumn;
+import static io.trino.plugin.hive.HiveType.HIVE_INT;
+import static io.trino.spi.type.BigintType.BIGINT;
+import static io.trino.testing.QueryAssertions.assertEqualsIgnoreOrder;
+
+public class TestHiveFileSystemS3SelectPushdownWithSplits
+        extends AbstractTestHiveFileSystemS3
+{
+    private String host;
+    private int port;
+    private SchemaTableName tableWith3MBSplits;
+
+    @Parameters({
+            "hive.hadoop2.metastoreHost",
+            "hive.hadoop2.metastorePort",
+            "hive.hadoop2.databaseName",
+            "hive.hadoop2.s3.awsAccessKey",
+            "hive.hadoop2.s3.awsSecretKey",
+            "hive.hadoop2.s3.writableBucket",
+            "hive.hadoop2.s3.testDirectory",
+    })
+    @BeforeClass
+    public void setup(String host, int port, String databaseName, String awsAccessKey, String awsSecretKey, String writableBucket, String testDirectory)
+    {
+        this.host = host;
+        this.port = port;
+        this.database = databaseName;
+        this.awsAccessKey = awsAccessKey;
+        this.awsSecretKey = awsSecretKey;
+        this.writableBucket = writableBucket;
+        this.testDirectory = testDirectory;
+        this.tableWith3MBSplits = new SchemaTableName(database, "trino_s3select_test_csv_scan_range_pushdown");
+
+        // Initial setup is needed for running the inherited tests
+        super.setup(host, port, database, awsAccessKey, awsSecretKey, writableBucket, testDirectory, true);
+    }
+
+    @DataProvider(name = "testSplitSize")
+    public static Object[][] splitSizeParametersProvider()
+    {
+        return new Object[][] {{3, 2}, {50, 30}};
+    }
+
+    @Test(dataProvider = "testSplitSize")
+    public void testQueryPushdownWithSplitSize(int maxSplitSizeMB, int maxInitialSplitSizeMB)
+            throws Exception
+    {
+        HiveConfig config = new HiveConfig()
+                .setS3SelectPushdownEnabled(true)
+                .setMaxSplitSize(DataSize.of(maxSplitSizeMB, MEGABYTE))
+                .setMaxInitialSplitSize(DataSize.of(maxInitialSplitSizeMB, MEGABYTE));
+
+        // Setup update is required to update the configuration
+        super.setup(host, port, database, awsAccessKey, awsSecretKey, writableBucket, testDirectory, config);
```

Comment:
This is not ideal. Tests are allowed to be executed in parallel and through this them may interfere with each other. I wonder how much do you get from extending the `AbstractTestHiveFileSystem`. How much of boilerplate would it cost to avoid the inheritance? With a separate class you would be able to initialize it exactly how you needed without unnecessary intricate interactions between a parent class and a test class.

Reply:
Updated, please check. I added a wrapper class for the configuration needed in S3 Select tests, because similar initializations were needed for both JSON and scan range tests.